### DISCUSSION TITLE
WATCHTOWER_CLEANUP - doesn't clean

### DISCUSSION THREAD

**Discussion by Nabukodonosor** [containrrr/watchtower#1211] [Q&A]
So, I have this env set up to TRUE, but it doesn't delete old images after the update. What's the problem here?

**Comment by Nabukodonosor** [comment]
Anyone?

**Reply by piksel** [reply]
Add your logs, preferably with `--debug` enabled, and how you are starting watchtower (command line or docker-compose). 

A few guesses; 
have you tried `true` instead of `TRUE`?
do you have additional quotes around the value in the declaration?

**Comment by Nabukodonosor** [comment]
docker-compose in Portainer.

[quote]have you tried true instead of TRUE?[/quote]
it is true. I just made it TRUE here to separate it from other regular words.

No quotes.

**Reply by piksel** [reply]
Add your logs, preferably with --debug enabled, and how you are starting watchtower (command line or docker-compose).

**Reply by Nabukodonosor** [reply]
Here you go:
https://pastebin.com/6LwJjaR2

**Reply by Nabukodonosor** [reply]
http://prntscr.com/26s4gah

**Reply by piksel** [reply]
It behaves as if you have not supplied `--cleanup`. Could you show your docker-file?

**Reply by Nabukodonosor** [reply]
docker run -d \
  --name watchtower \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -e WATCHTOWER_NOTIFICATIONS=email \
  -e WATCHTOWER_NOTIFICATION_EMAIL_FROM=watchtower@gmail.com \
  -e WATCHTOWER_NOTIFICATION_EMAIL_TO=myemail@gmail.com \
  -e WATCHTOWER_NOTIFICATION_EMAIL_SERVER=smtp.gmail.com \
  -e WATCHTOWER_NOTIFICATION_EMAIL_SERVER_USER=myemail@gmail.com \
  -e WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PASSWORD=mypassword \
  -e WATCHTOWER_CLEANUP=true \
  -e WATCHTOWER_NOTIFICATION_EMAIL_SERVER_PORT=587 \
  -e WATCHTOWER_NOTIFICATION_EMAIL_SUBJECTTAG=Docker Container Updates \
  -e WATCHTOWER_MONITOR_ONLY=true \
  -e WATCHTOWER_INCLUDE_STOPPED=Europe/Belgrade \
  -e TZ=true \
  containrrr/watchtower

**Reply by Nabukodonosor** [reply]
So, should I use --clean or WATCHTOWER_CLEANUP=true? I have never seen this --clean argument before.

**Reply by piksel** [reply]
No, I meant `--cleanup`. You can use either that or the env var.

You are using `WATCHTOWER_MONITOR_ONLY=true` though, so there should never be anything to clean up (since it's never updating anything).

**Reply by Nabukodonosor** [reply]
I know, but I got the mail that I have an update, I do the update manually, and after that...well, you've seen my screenshot. But I will try to remove that monitor only thing. But that's not a really good solution though.

**Reply by piksel** [reply]
`--cleanup` means "remove the old image after the update", so if there is no update performed, there is no cleanup.

If you just want a general "remove all images not being used", then you are looking for `docker image prune`.

**Reply by Nabukodonosor** [reply]
Cool, thanks!
